### PR TITLE
docs: restart events doc edit

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Review and edit of the restart events content in the _Deploying_ guide.
The content is now presented as a separate chapter.

- Adds an assembly for the content
- Splits the reference doc from the procedure
- Review edits to content

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/deploying/ref-operator-restart-events-fields.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+`reportingComponent`:: The reporting component is always `strimzi.io/cluster-operator` for Strimzi restart events.
+`source`:: `source` is an older version of `reportingComponent`. The reporting component is always `strimzi.io/cluster-operator` for Strimzi restart events.
+`type`:: The event type, which is either `Warning` or `Normal`. For Strimzi restart events, the type is `Normal`.
+
+.Additional event fields for Kubernetes 1.25 and later
+
+`regarding.kind`:: Same as `involvedObject.kind`.
+`regarding.namespace`:: Same as `involvedObject.namespace`.
+`regarding.name`:: Same as `involvedObject.name`.
+`regarding.uid`:: Same as `involvedObject.uid`.
```
Comment (PaulRMellor) [inline documentation/modules/deploying/ref-operator-restart-events-fields.adoc:29]: @LiamClarkeNZ  this was showing as _Same as  `involvedObject.name`_. Assumed it should be  `involvedObject.uid`

**Review by scholzj — APPROVED** [review]

File: documentation/modules/deploying/ref-operator-restart-events-reasons.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+`ConfigChangeRequiresRestart`:: Some Kafka configuration properties are changed dynamically, but others require that the broker be restarted.
+`CustomListenerCaCertChanged`:: The CA certificate used to secure the Kafka network listeners has changed, and the pod is restarted to use it.
+`FileSystemResizeNeeded`:: The file system size has been increased, and a restart is needed to apply it.
+`JbodVolumesChanged`:: A disk was added or removed from the Kafka volumes, and a restart is needed to apply the change.
+`ManualRollingUpdate`:: A user annotated the pod, or the `StatefulSet`` or `StrimziPodSet` set it belongs to, to trigger a restart.
+`PodForceRestartOnError`:: An error occurred that requires a pod restart to rectify.
+`PodHasOldGeneration`:: The `StatefulSet` that the pod is a member of has been updated, so the pod needs to be recreated.
+`PodHasOldRevision`:: The `StrimziPodSet` that the pod is a member of has been updated, so the pod needs to be recreated.
+`PodStuck`:: The pod is still pending, and is either unschedulable or not scheduled, so the operator has restarted the pod as a final attempt to get it running.
```
Comment (joaedwar) [inline documentation/modules/deploying/ref-operator-restart-events-reasons.adoc:27]: I don't think _unscheduable_ is not in the MW dictionary.
```suggestion
`PodStuck`:: The pod is still pending, and either is not scheduled or can not be scheduled, so the operator has restarted the pod in a final attempt to get it running.
```

File: documentation/modules/deploying/ref-operator-restart-events-reasons.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+`PodUnresponsive`:: Strimzi was unable to connect to the pod, which can indicate a broker not starting properly, so the operator will restart it to try to resolve the issue.
```
Comment (joaedwar) [inline documentation/modules/deploying/ref-operator-restart-events-reasons.adoc:28]:
```suggestion
`PodUnresponsive`:: Strimzi was unable to connect to the pod, which can indicate a broker not starting correctly, so the operator will restart it in an attempt to resolve the issue.
```

File: documentation/modules/deploying/ref-operator-restart-events-fields.adoc (hunk 1 above)
Comment (LiamClarkeNZ) [reply documentation/modules/deploying/ref-operator-restart-events-fields.adoc:29]: Nice catch :)

**Review by LiamClarkeNZ — APPROVED** [review]

**Comment by LiamClarkeNZ** [comment]
Looks great, thanks for looking into this @PaulRMellor

**Pull request merged**